src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -312,7 +312,7 @@ impl<'a> Resolver<'a> {
                             Ident::new(keywords::SelfLower.name(), new_span)
                         ),
                         kind: ast::UseTreeKind::Simple(
-                            Some(Ident::new(keywords::Underscore.name().gensymed(), new_span)),
+                            Some(Ident::new(Name::gensym("__dummy"), new_span)),
```

Comment:
Doesn't this shift the problem to the possible existence of `__dummy` somewhere in the namespace?

Reply:
No, it's still a gensym, so it doesn't conflict with anything locally + it's not written into metadata since it's no longer an underscore.
(Previously it wasn't written due to the `binding.vis != ty::Visibility::Invisible` condition, but I removed it to simplify things a bit.)